### ISSUE TITLE
Example of generate big integer not working

### ISSUE THREAD

**Issue by Escapingbug** [rust-num/num-bigint#14]
OK, I don't know what I'm wrong about this. But the example from the documentation seems not able to work.
```
extern crate rand;
extern crate num;

use num::bigint::{BigInt, RandBigInt, ToBigInt};

#[test]
fn testme() {

    let mut rng = rand::thread_rng();
    let a = rng.gen_bigint(1000);

    let low = -10000.to_bigint().unwrap();
    let high = 10000.to_bigint().unwrap();
    let b = rng.gen_bigint_range(&low, &high);

    // Probably an even larger number.
    println!("{}", a * b);
}
```
I think the trait should be covered already. But I got error:
```
error[E0599]: no method named `gen_bigint` found for type `rand::ThreadRng` in the current scope
   --> src/rsa.rs:111:17
    |
111 |     let a = rng.gen_bigint(1000);
    |                 ^^^^^^^^^^
    |
    = note: the method `gen_bigint` exists but the following trait bounds were not satisfied:
            `rand::ThreadRng : num::bigint::RandBigInt`

error[E0599]: no method named `gen_bigint_range` found for type `rand::ThreadRng` in the current scope
   --> src/rsa.rs:115:17
    |
115 |     let b = rng.gen_bigint_range(&low, &high);
    |                 ^^^^^^^^^^^^^^^^
    |
    = note: the method `gen_bigint_range` exists but the following trait bounds were not satisfied:
            `rand::ThreadRng : num::bigint::RandBigInt`

error: aborting due to 2 previous errors
```

Is the usage changed or I'm doing it the wrong way?

**Comment by Escapingbug** [comment]
I think I know why it happened. I changed my rand version to `0.3.14` and it is working now. But I see a PR here to relax the rand version? Won't that be a problem?

**Comment by cuviper** [comment]
My thought was that rand version flexibility would be nicer, but maybe that will cause headaches of its own if different parts of the dependency graph actually try to use different rand. I intend to bump semver soon anyway, so I guess we could just go to strictly rand 0.4 then.

**Comment by cuviper** [comment]
The rand PR is now merged and published, and also rand 0.3.22 now re-exports rand 0.4 so they're compatible anyway.
